syntax = "proto3";

enum ProtoAbilityType {
    ProtoClimaxTrigger = 0;
    ProtoGlobal = 1;
    ProtoCard = 2;
    ProtoRuleAction = 3;
    ProtoDelayed = 4;
}

enum ProtoOwner {
    ProtoPlayer = 0;
    ProtoOpponent = 1;
}

message ProtoAbility {
    ProtoAbilityType type = 1;
    string zone = 2;
    string card_code = 3;
    sint32 card_id = 4;
    sint32 ability_id = 5;
    bytes ability = 6;
    uint32 unique_id = 7;
}

enum ProtoCardBoolAttribute {
    ProtoCannotFrontAttack = 0;
    ProtoCannotSideAttack = 1;
    ProtoCannotBecomeReversed = 2;
    ProtoCannotMove = 3;
    ProtoSideAttackWithoutPenalty = 4;
    ProtoCannotStand = 5;
    ProtoCannotBeChosen = 6;
    ProtoPlayWithoutColorRequirement = 7;
}

enum ProtoPlayerAttribute {
    ProtoCannotPlayBackups = 0;
    ProtoCannotPlayEvents = 1;
    ProtoCharAutoCannotDealDamage = 2;
}
